### DISCUSSION TITLE
Slow query log?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#5879] [Q&A]
We had an incident on pockethost.io today where a user's app went viral and one of his tables ballooned to 50k records with no indexes. Full table scan queries approached 1000ms.

If possible, I'd like a way to surface this kind of issue via an alert on PocketHost. 

What approaches are available to track and report slow queries?

**Comment by ganigeorgiev** [comment]
There is no builtin mechanism for this at the moment and tracking all individual queries all the time could eventually make things worse.

An alternative could be to track routes execution, aka. to register a middleware and call `started := time.Now()` or similar before the `next` action and after it completes do something based on the `time.Now().Sub(started)` difference (this is usually how the `execTime` prop of the activity logs works).

There are plans to develop an "Indexes advisory" settings section where you'll be able to temporary enable query execution tracking and ideally be able to suggest indexes but this remains a low priority for now.

Note that the number of records is not always the reason for the slow queries and 50k is not really that much. You could also check some of the notes in another recent discussion https://github.com/pocketbase/pocketbase/discussions/5847.
